Lib/test/test_ssl.py

```diff
@@ -2800,6 +2813,44 @@ def test_echo(self):
                 'Cannot create a client socket with a PROTOCOL_TLS_SERVER context',
                 str(e.exception))
 
+    @unittest.skipUnless(support.Py_GIL_DISABLED, "test is only useful if the GIL is disabled")
+    def test_ssl_in_multiple_threads(self):
+        # See GH-124984
+        threads = []
+
+        USE_SAME_TEST_CONTEXT = True
```

Comment:
Missing a `global` statement so this doesn't do anything. Also, let's use a finally block to switch this back.

Reply:
Ah, right. I'll deal with this in the morning.

Reply:
Fixed.